packages/flutter/test/material/theme_test.dart

```diff
@@ -47,6 +47,74 @@ void main() {
     expect(Theme.of(tester.element(find.text('menuItem'))).brightness, equals(Brightness.dark));
   });
 
+  group('Theme.brightnessOf', () {
+    testWidgets('return correct brightness when just media query is given', (
+      WidgetTester tester,
+    ) async {
+      await tester.pumpWidget(
+        MediaQuery(
+          data: const MediaQueryData(platformBrightness: Brightness.dark),
+          child: Container(),
+        ),
+      );
+
+      expect(Theme.brightnessOf(tester.element(find.byType(Container))), equals(Brightness.dark));
+    });
+
+    testWidgets('return correct brightness with overriding theme brightness over media query', (
+      WidgetTester tester,
+    ) async {
+      await tester.pumpWidget(
+        Theme(
+          data: ThemeData(brightness: Brightness.light),
+          child: MediaQuery(
```

Comment:
Should the MediaQuery be above the Theme, for consistency with the real widget tree in MaterialApp?